src/solve.ml

```diff
@@ -114,11 +127,14 @@ let generate_bowtie = curry3 @@ memoize @@ fun (spec, m1, m2) ->
         ["))"]
     in
     
+    let post = if String.equal (postcond_args_list "12" []) "" then [] else [ sp "   (postcondition %s)" (postcond_args_list "12" [])] in
```

Comment:
Is this only holding true of the `12` version of the variables? I guess you could argue that we must have that `states_equal` of the `12` and `21` versions, and that should imply that the postcondition holding in just one is sufficient for it holding in both.

Reply:
(Probably just add a note saying this as a comment.)